### PR TITLE
Filtering

### PR DESCRIPTION
**Pull request by wejendorp**
-   Moved full-text index generation into functions `Util.purge_search` and `Util.update_search`, in order to avoid repopulating entire database when indexes are changed.
-    Added an `AppModel.search_todict` function, which returns all matches as a `dict list`.
-    Added working Study and Disease searches (on all fields).

### CODE REVIEW

**Comment by wejendorp** [comment]
hey @jensraaby and @paddie, could you take a look at this? Just test the search fields on studylistview and diseaselistview.

**Comment by paddie** [comment]
On it boyo!

**Comment by paddie** [comment]
der er ikke noget søge felt på nogen af vinduerne; hvordan tester jeg?

**Comment by wejendorp** [comment]
``` python
from Util import update_search
update_search()
```

in the interactive console should fix you right up :)

**Comment by paddie** [comment]
ahh, skulle bare køre populate så kom der et felt frem.. Eller også havde det været der hele tiden.. who knows? :)

Det virker, den crasher på en tom streng dog. Måske en måde at vise alle diseases/studies igen efter en søgning?

**Comment by wejendorp** [comment]
God idé. Det adder jeg.

**Comment by paddie** [comment]
Kører nice ellers!

**Comment by wejendorp** [comment]
That ought to do it.

**Comment by paddie** [comment]
crasher stadig ved blank søgning

**Comment by paddie** [comment]
hvis du kunne placere søgning i toppen af hvert vindue, ala collections - det ville være nemmere at se.

**Comment by paddie** [comment]
Ha, på disease view er "None" sat some value, hvis den er sat så overskrive den placeholder..

File: templates/diseaselistrender.html [hunk 1]
```diff
@@ -1,4 +1,12 @@
+<form method="GET" class="pull-right">
+    <input class="input search-query" placeholder="Disease search"
+    name="filter" value="{{filter if filter is defined}}" />
```
Comment (paddie) [inline templates/diseaselistrender.html:4]: filter er sat til `None` ved første GET

**Comment by wejendorp** [comment]
Har ikke kunnet genskabe din None fejl, men nu er der ekstra safeguards på.
Har også flyttet søgeboxen så selv DU kan få øje på den ;)
Kan vi snart merge this shit? :P

**Comment by paddie** [comment]
jeg tror det skete da jeg forsøgte at søge på et tomt felt, resultatet blev at filter="None" istedet for Python forståelsen af None.

**Comment by paddie** [comment]
GYS: fundet en episk fejl i vores GWAS import, som nok kommer til at påvirke din indexering af studies. Kan du checke mit andet pull-request ud?

**Comment by jensraaby** [comment]
Hvad er status på denne pullrequest?

**Comment by paddie** [comment]
Samme pointe, da han bruger index til det, så vil det blive påvirket
af det andet pull-request

Hilsen

Patrick-Ranjit D. Madsen

Sent from my mobile. Please excuse any and all typos, errors,
formatting, lack of punctuation or short responses.

On 01/06/2012, at 20.25, jensraaby
reply@reply.github.com
wrote:

**Comment by wejendorp** [comment]
I cleaned up the commits while rebasing, its ready for merging.

**Comment by jensraaby** [comment]
Doesn't work for me when I search for 'cancer' on the disease page. Running update_search() does this:

Traceback (most recent call last):
  File "C:\Program Files (x86)\Google\google_appengine\google\appengine\ext\admin__init__.py", line 320, in post
    exec(compiled_code, globals())
  File "<string>", line 3, in <module>
  File "C:\Users\Laursen\Dropbox\Computer Science\PerseQ\Util.py", line 305, in update_search
    AddStudyDocument(study)
  File "C:\Users\Laursen\Dropbox\Computer Science\PerseQ\Util.py", line 41, in AddStudyDocument
    search.TextField(name='disease_trait', value=','.join([s.name for s in study.diseases])),
TypeError: sequence item 0: expected string, instancemethod found

**Comment by jensraaby** [comment]
So, now the update_search method works, but the actual disease filtering isn't working at all:

  File "C:\Users\Laursen\Dropbox\Computer Science\PerseQ\controllers\gwasController.py", line 22, in get
    diseases = Disease.search_todict('"'+myfilter+'"')
  File "C:\Users\Laursen\Dropbox\Computer Science\PerseQ\models\Application.py", line 27, in search_todict
    for field in doc.fields:
AttributeError: 'SearchResult' object has no attribute 'fields'

**Comment by wejendorp** [comment]
Need more info to replicate.

**Comment by jensraaby** [comment]
steps> checkout filtering branch. purge, populate, update_search
load http://localhost:8080/diseases/?filter=cancer
error 500 as above

**Comment by jensraaby** [comment]
Same goes for http://localhost:8080/studies/?filter=cancer

**Comment by wejendorp** [comment]
Could it be that your SDK is not up to date, cause I can't replicate it?

**Comment by jensraaby** [comment]
I'll try that. It is 1.6.4 at the moment

Jens

**Comment by paddie** [comment]
Should be 1.6.6

Hilsen

Patrick-Ranjit D. Madsen

Sent from my mobile. Please excuse any and all typos, errors,
formatting, lack of punctuation or short responses.

On 05/06/2012, at 14.41, jensraaby
reply@reply.github.com
wrote:

> I'll try that. It is 1.6.4 at the moment
> 
> Jens
>

**Comment by jensraaby** [comment]
Today I found out how to do post/get testing, but am really rather confused about why the search doesn't work (no error now, but i get no results for any queries like 'cancer'. This is why we needed tests for things merged in to master). I am probably going to make a mess of a presentation unless one of you can have a working demo ready tomorrow. 
I sorry this week is turning out a bit messy from my side. It's not going to get any easier so im relying on you guys to help keep up the momentum...

Jens

**Comment by paddie** [comment]
I should probably mention that my hard drive just took a nosedive..
Happened 2 days ago and I've been going to work to get stuff done ever
since. And because of grundlovsdag I'm not getting any of the parts
I've ordered before tomorrow :/

I'll drop by work tonight and look at the slides.

Hilsen

Patrick-Ranjit D. Madsen

Sent from my mobile. Please excuse any and all typos, errors,
formatting, lack of punctuation or short responses.

On 05/06/2012, at 19.23, jensraaby
reply@reply.github.com
wrote:

> Today I found out how to do post/get testing, but am really rather confused about why the search doesn't work (no error now, but i get no results for any queries like 'cancer'. This is why we needed tests for things merged in to master). I am probably going to make a mess of a presentation unless one of you can have a working demo ready tomorrow.
> I sorry this week is turning out a bit messy from my side. It's not going to get any easier so im relying on you guys to help keep up the momentum...
> 
> Jens
>

**Comment by jensraaby** [comment]
OK seems we've got an unlucky streak this week...

**Comment by paddie** [comment]
Alright, I have no issue searching for cancer.. I don't know what the issue is with Jens' machine.. Maybe it's your python build?

**Comment by paddie** [comment]
If it's ready let's get this pull'ed!

**Comment by jensraaby** [comment]
The issue with my machine is either python or gae. I can't run populate for some reason involving overwriting a file. If search works on other machines then that's great :)

Jens

**Comment by paddie** [comment]
Have you run `purge()` first?

**Pull request merged**